python/create_service_diagnostics_bundle.py

```diff
@@ -138,99 +52,259 @@ def parse_args() -> dict:
-    ).create()
+    @property
+    def marathon_info(self):
+        if self._marathon_app is None:
+            self._marathon_app = self.get_marathon_app(self._service_name)
+            self.validate_package_name(
+                self._package_name, self._marathon_app, self.service_name)
 
-    return rc
+        return self._marathon_app
+
+    @property
+    def package_name(self) -> str:
+        if self._package_name is None:
+            self.load_args()
+
+        return self._package_name
+
+    @property
+    def service_name(self) -> str:
+        if self._service_name is None:
+            self.load_args()
+
+        return self._service_name
+
+    @property
+    def bundles_directory(self) -> str:
+        if self._bundles_directory is None:
+            self.load_args()
+
+        return self._bundles_directory
+
+    @property
+    def should_prompt_user(self) -> bool:
+        if self._should_prompt_user is None:
+            self.load_args()
+
+        return self._should_prompt_user
+
+    @property
+    def package_version(self) -> str:
+        if self._package_version is None:
+            marathon_app = self.marathon_info
+            self._package_version = "n/a"
+            if marathon_app is not None:
+                self._package_version = marathon_app.get("labels", {}).get(
+                    "DCOS_PACKAGE_VERSION")
+
+        return self._package_version
+
+    @property
+    def dcos_version(self) -> str:
+        if self._dcos_version is None:
+            self._dcos_version = self.clusters[0]["version"]
+        return self._dcos_version
+
+    @property
+    def cluster_url(self) -> str:
+        if self._cluster_url is None:
+            self._cluster_url = self.clusters[0]["url"]
+        return self._cluster_url
+
+    @property
+    def service_diagnostics_version(self) -> str:
+        if self._service_diagnostics_version is None:
+            __location__ = os.path.realpath(
+                os.path.join(os.getcwd(), os.path.dirname(__file__)))
+
+            with open(os.path.join(__location__, 'VERSION'), 'r') as f:
+                self._service_diagnostics_version = f.readline()
+
+        return self._service_diagnostics_version
+
+
+def main() -> int:
+    try:
+        config = Configuration()
+        config.check_authentication()
+        config.check_cluster_state()
+
+        s = "  {:18}{}"
+        print("Will create bundle for:")
+        print(s.format("Package:", config.package_name))
+        print(s.format("Package version:", config.package_version))
+        print(s.format("Service name:", config.service_name))
+        print(s.format("DC/OS version:", config.dcos_version))
+        print(s.format("Cluster URL:", config.cluster_url))
+
+        if config.should_prompt_user:
+            answer = input("\nProceed? [Y/n]: ")
+            if answer.strip().lower() not in ["yes", "y", ""]:
+                return 0
+
+        rc, _ = FullBundle(
```

Comment:
Since this needs so many properties of the config, why not just pass the config object?

Reply:
There will be a cross dependency in this case:
```
from full_bundle import eFullBundle
```
in create_service_diagnostics_bundle.py and FullBundle will require import `Configuration`
I can generate *args and send them as params and pass in FullBundle constructor

Reply:
Sounds like something easily fixed by moving the `Configuration` class to a separate file? I guess it's big enough for that to make sense? The whole point of me asking for the refactor was to change this call to not require so many parameters :-)

Reply:
done, please check implementation